docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,266 @@
+---
+page_title: "Migration Guide: Transition out of Serverless Instances and Shared-tier clusters"
+---
+
+# Migration Guide: Transition out of Serverless Instances and Shared-tier clusters
+
+The goal of this guide is to help users transition from Serverless Instances and Shared-tier clusters (M2/M5) to Free, Flex or Dedicated Clusters. 
+
+Starting in March 2025, all Shared-tier clusters (in both `mongodbatlas_cluster` and `mongodbatlas_advanced_cluster`) will automatically convert to Flex clusters. Similarly, all Serverless instances (`mongodb_serverless_instance`) will be converted into Free/Flex/Dedicated clusters, [depending on your existing configuration](https://www.mongodb.com/docs/atlas/flex-migration/). 
+These conversions will result in configuration drift in Terraform. 
```

Comment:
and also we should make clear that by January 2026 they wil stop working

Reply:
As per my understanding, Serverless and Shared-tier will be deprecated in January 2025, and EOL by Jan 2026, and we should share that serverless and shared tier will stop working in January 2025 along the deprecations. If we share this here, I think it might be confusing because the resources will not be deprecated yet